<a href="https://colab.research.google.com/github/JuanManuelLopezP/Anal-tica/blob/main/Reto_1_Juan_Manuel.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import confusion_matrix

In [ ]:
nxl= '/content/1. BD1_Longitudinal Survey.xlsx'
XDB=pd.read_excel(nxl,sheet_name=0)
XDB = XDB.dropna()
XDB=XDB[['Age','Mother_Edu','Father_Edu','Height','Weight','Income','Self_Esteem','FamilySize','Urban']]
XD=np.array(XDB[['Age','Mother_Edu','Father_Edu','Height','Weight','Income','Self_Esteem','FamilySize']])
yd=np.array(XDB['Urban'])
XDB=np.array(XDB)
print(XDB)

mnd=GaussianNB()
mnd.fit(XD,yd)
ydp=mnd.predict(XD)

[[20.  5.  8. ... 16.  5.  1.]
 [19. 12. 12. ... 27.  4.  1.]
 [15. 12. 12. ... 26.  3.  1.]
 ...
 [21. 16. 12. ... 30.  1.  1.]
 [21. 16. 16. ... 27.  1.  1.]
 [21.  8.  8. ... 19.  1.  0.]]


In [ ]:
np.sum(yd)/len(yd)
1-np.sum(yd)/len(yd)

0.20640506774590883

In [ ]:
mnb=GaussianNB()
mnb.fit(XD,yd)
ydp=mnb.predict(XD)

In [ ]:
u=mnb.theta_
sigma=np.sqrt(mnb.var_)

In [ ]:
mnb=GaussianNB()
mnb.fit(XD,yd)
ydp=mnb.predict(XD)
u=mnb.theta_
sigma=np.sqrt(mnb.var_)


In [ ]:
LI=u-sigma
LS=u+sigma
print("La media de las variables por categoría")
print(u)
print("El limite inferior de la variación de las variables por categoría es")
print(LI)
print("El limite superior de la variación de las variables por categoría es")
print(LS)

La media de las variables por categoría
[[1.83810742e+01 1.07612958e+01 1.03324808e+01 6.72148338e+01
  1.48857630e+02 3.03672506e+04 2.20272805e+01 4.74168798e+00]
 [1.83569845e+01 1.11769401e+01 1.12565410e+01 6.69031042e+01
  1.44760976e+02 3.34278929e+04 2.26004435e+01 4.68780488e+00]]
El limite inferior de la variación de las variables por categoría es
[[ 1.58996876e+01  7.58892579e+00  6.46311334e+00  6.31369799e+01
   1.18746964e+02 -1.33465823e+03  1.78344685e+01  2.16621834e+00]
 [ 1.58811112e+01  7.79013864e+00  7.15097738e+00  6.26897933e+01
   1.14669654e+02 -2.69533531e+03  1.84154526e+01  2.24996381e+00]]
El limite superior de la variación de las variables por categoría es
[[2.08624608e+01 1.39336659e+01 1.42018483e+01 7.12926877e+01
  1.78968296e+02 6.20691595e+04 2.62200925e+01 7.31715762e+00]
 [2.08328577e+01 1.45637416e+01 1.53621047e+01 7.11164151e+01
  1.74852297e+02 6.95511211e+04 2.67854344e+01 7.12564594e+00]]


In [ ]:
LI_age = LI[0, 0]
LS_age = LS[0, 0]

print(f"El intervalo de la variación de la edad es: [{LI_age:.2f}, {LS_age:.2f}]")

El intervalo de la variación de la edad es: [15.90, 20.86]


In [ ]:
LI_Height = LI[0, 3]
LS_Height = LS[0, 3]

print(f"El intervalo de la variación de la altura es: [{LI_Height:.2f}, {LS_Height:.2f}]")

El intervalo de la variación de la altura es: [63.14, 71.29]


In [ ]:
LI_Weight = LI[0, 4]
LS_Weight = LS[0, 4]

print(f"El intervalo de la variación de el peso es: [{LI_Weight:.2f}, {LS_Weight:.2f}]")

El intervalo de la variación de el peso es: [118.75, 178.97]


In [ ]:
cm=confusion_matrix(yd,ydp)
print(cm)


[[   2 1171]
 [   4 4506]]


In [ ]:
from re import S
VN=cm[0,0]
FP=cm[0,1]
FN=cm[1,0]
VP=cm[1,1]

Exactitud=(VP+VN)/len(XD)
print("La exactitud es:",Exactitud)

TE=(FN+FP)/len(XD)
print("El Tasa de Error es:",TE)

Sensibilidad=VP/(VP+FP)
print("La sensibilidad es:",Sensibilidad)

Especificidad=VN/(VN+FN)
print("La especificidad es:",Especificidad)

La exactitud es: 0.7932430054548654
El Tasa de Error es: 0.2067569945451346
La sensibilidad es: 0.793729082261758
La especificidad es: 0.3333333333333333


In [ ]:
Xun=np.array([[15,12,12,65,131,25000,26,3]])  #Misma persona del excel
ydun=mnd.predict(Xun)
print(ydun)

[1.]


**Pregunta 1**
La cantitdad total de datos de la base es  de 12678 datos

**Pregunta 2**
El porcentaje de datos que pertenecena a las categorías de clasificación equivale al 79.35% y los que no pertenecen a esta equivalen al siguiente porcentaje 20.64%

**Pregunta 3**
El intervalo de la variación de la edad es: [15.90, 20.86]

**Pregunta 4**
El intervalo de la variación de la altura es: [63.14, 71.29]

**Pregunta 5**
El intervalo de la variación de el peso es: [118.75, 178.97]

**Pregunta 6**
La excatidud del modelo es de 79.32%

**Pregunta 7**
La tasa de error del modelo es de 20.67%

**Pregunta 8**
La sensibilidad del modelo es de 79.37%

**Pregunta 9**
La especificidad del modelo es de 33.33%

**Pregunta 10**
Para una persona con las siguientes características 15,12,12,65,131,25000,26,3 (Edad, Educación de madre y padre, altura, peso, ingresos, autoestima y tamaño de la familia) el modelo demuestra que esta vive en el área urbana

**Analisis**

Con el modelo anteriormente presentado, podemos observar que la mayoría de personas que se encuentran en la base de datos, viven en el sector urbano, una de las razones de esto pueden ser el tamaño de las familias que en el caso de las personas que viven en zona no urbana, estos presentan número mas grandes, en cuanto a las métricas, podemos inferir que gracias al modelo, este tuvo una sensibilidad 79.37% de certeza. En general, podemos afirmar que este modelo sirve ya que presenta un 79.32% de exactitud a la hora de evaluar a los candidatos y una tasa de error de 20.67%. Finalmente con la persona que se escogió para probar el modelo, podemos ver que el resultado arrojado por el modelo es de 1 lo que equivale a que el individio vive en zona urbana y al comparalo con la base de datos donde este ya tiene su clasificación, se encuentra que la al igual que el modelo, el sujeto de prueba vive efectivamente en zona urbana.